sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeHealthcareEntitiesOperation.cs

```diff
@@ -374,15 +308,46 @@ Page<AnalyzeHealthcareEntitiesResultCollection> NextPageFunc(string nextLink, in
                 }
             }
 
-            return PageableHelpers.CreateEnumerable(_ => _firstPage, NextPageFunc);
+            return PageableHelpers.CreateAsyncEnumerable(_ => Task.FromResult(_firstPage), NextPageFunc);
         }
 
-        private void ValidateOperationStatus()
+        async ValueTask<OperationState<AsyncPageable<AnalyzeHealthcareEntitiesResultCollection>>> IOperation<AsyncPageable<AnalyzeHealthcareEntitiesResultCollection>>.UpdateStateAsync(bool async, CancellationToken cancellationToken)
         {
-            if (!HasCompleted)
-                throw new InvalidOperationException("The operation has not completed yet.");
-            if (!HasValue)
-                throw _requestFailedException;
+            var response = async
+                ? await _serviceClient.HealthStatusAsync(new Guid(Id), null, null, _showStats, cancellationToken).ConfigureAwait(false)
+                : _serviceClient.HealthStatus(new Guid(Id), null, null, _showStats, cancellationToken);
+
+            // Add lock to avoid race condition?
```

Comment:
The consumer of this API won't be able to read all values atomically anyway.

Reply:
The concern would be in case the operation ended up in an invalid state because of parallel calls (some variables from call A, some from call B). This behavior was already present before the new `OperationInternal` class, so it's more like a Text Analytics discussion. Not directly related to this PR.

Reply:
Yeah, let's keep it as is for now.